beacon-chain/types/block_test.go

```diff
@@ -1,59 +1,74 @@
 package types
 
 import (
-	"reflect"
 	"testing"
 
+	"github.com/prysmaticlabs/prysm/beacon-chain/params"
 	pb "github.com/prysmaticlabs/prysm/proto/beacon/p2p/v1"
+	"github.com/sirupsen/logrus"
 )
 
-func TestBlock(t *testing.T) {
-	data := &pb.BeaconBlock{
-		ParentHash:            []byte{0},
-		SlotNumber:            0,
-		RandaoReveal:          []byte{0},
-		Attestations:          []*pb.AttestationRecord{},
-		PowChainRef:           []byte{0},
-		ActiveStateHash:       []byte{0},
-		CrystallizedStateHash: []byte{0},
-	}
-	block := &Block{data}
-	block.ParentHash()
-	block.PowChainRef()
-	block.RandaoReveal()
-	block.ActiveStateHash()
-	block.CrystallizedStateHash()
-	newBlockNil := NewBlock(nil)
-	block.data.Timestamp = newBlockNil.data.Timestamp
-	if !reflect.DeepEqual(block, newBlockNil) {
-		t.Errorf("mismatched blocks: wanted %v, received %v", block, newBlockNil)
-	}
-	data = &pb.BeaconBlock{SlotNumber: 1}
-	block = &Block{data}
-	block.data.Timestamp = newBlockNil.data.Timestamp
-	if !reflect.DeepEqual(block, NewBlock(data)) {
-		t.Errorf("mismatched blocks: wanted %v, received %v", block, NewBlock(data))
-	}
-	emptyBlock := &Block{}
-	if _, err := emptyBlock.Marshal(); err == nil {
-		t.Error("marshal with empty data should fail")
-	}
-	if _, err := emptyBlock.Hash(); err == nil {
-		t.Error("hash with empty data should fail")
-	}
-	if _, err := block.Timestamp(); err != nil {
-		t.Errorf("well formatted timestamp should not throw an error, received %v", err)
-	}
-	if _, err := block.Hash(); err != nil {
-		t.Errorf("hashing with data should not fail, received %v", err)
-	}
-	if !reflect.DeepEqual(block.data, block.Proto()) {
-		t.Errorf("inner block data did not match proto: received %v, wanted %v", block.Proto(), block.data)
-	}
-	if block.AttestationCount() != 0 {
-		t.Errorf("mismatched attestation count: wanted 0, received %v", block.AttestationCount())
-	}
-	if _, err := NewGenesisBlock(); err != nil {
-		t.Errorf("creating genesis block should not throw error, received %v", err)
+func TestGenesisBlock(t *testing.T) {
+	b1, err1 := NewGenesisBlock()
+	b2, err2 := NewGenesisBlock()
+	if err1 != nil || err2 != nil {
+		t.Fatalf("failed to instantiate genesis block: %v %v", err1, err2)
+	}
+
+	h1, err1 := b1.Hash()
+	h2, err2 := b2.Hash()
+	if err1 != nil || err2 != nil {
+		t.Fatalf("failed to hash genesis block: %v %v", err1, err2)
+	}
+
+	if h1 != h2 {
+		t.Fatalf("genesis block hash should be identical: %x %x", h1, h2)
+	}
+
+	b3 := NewBlock(nil)
+	h3, err3 := b3.Hash()
+	if err3 != nil {
+		t.Fatalf("failed to hash genesis block: %v", err3)
+	}
+
+	if h1 == h3 {
+		t.Fatalf("genesis block and new block should not have identical hash: %x", h1)
+	}
+}
+
+func TestBlockValidity(t *testing.T) {
+	logrus.SetLevel(logrus.DebugLevel)
```

Comment:
Can you do this in `init() { .. }`? This is the pattern we use elsewhere, i think

Reply:
Ah, I knew something like this existed, thanks!